code/game/objects/structures/electricchair.dm

```diff
@@ -1,52 +1,132 @@
+#define ELECTRIC_CHAIR_WAIT_TIME 5
+
 /obj/structure/chair/e_chair
 	name = "electric chair"
 	desc = "Looks absolutely SHOCKING!"
 	icon_state = "echair0"
-	var/obj/item/assembly/shock_kit/part = null
 	var/last_time = 1
 	item_chair = null
 
 /obj/structure/chair/e_chair/Initialize()
 	. = ..()
-	add_overlay(mutable_appearance('icons/obj/chairs.dmi', "echair_over", MOB_LAYER + 1))
+	if(!stored_kit)
+		stored_kit = new(src)
+		stored_kit.master = src
+		AddComponent(/datum/component/electrified_chair, stored_kit)
 
 /obj/structure/chair/e_chair/attackby(obj/item/W, mob/user, params)
 	if(W.tool_behaviour == TOOL_WRENCH)
 		var/obj/structure/chair/C = new /obj/structure/chair(loc)
 		W.play_tool_sound(src)
 		C.setDir(dir)
-		part.forceMove(loc)
-		part.master = null
-		part = null
+		stored_kit.forceMove(loc)
+		stored_kit.master = null
+		stored_kit = null
 		qdel(src)
 
-/obj/structure/chair/e_chair/proc/shock()
-	if(last_time + 50 > world.time)
-		return
-	last_time = world.time
 
-	// special power handling
-	var/area/A = get_area(src)
-	if(!isarea(A))
-		return
-	if(!A.powered(AREA_USAGE_EQUIP))
+/datum/component/electrified_chair
+	///the chair that we are attached to
+	var/obj/structure/chair/parent_chair
+	///the shock kit attached to parent_chair
+	var/obj/item/assembly/shock_kit/used_shock_kit
```

Comment:
a shock kit shouldn't be a necesity for this component to work

Reply:
a shock kit is no longer necessary although currently nothing actually calls AddComponent with anything other than a shock kit